AudioContextInterruptedState/explainer.md

```diff
@@ -0,0 +1,212 @@
+3. The screen gets locked. At this moment, based on the the Audio Session API type, the UA should interrupt the `AudioContext`, and the context's state should now be `"interrupted"`.
+4. The screen gets unlocked. The UA should lift the interruption and the `AudioContext` should transition to the `"running"` state.
+
+```js
+<!DOCTYPE html>
+<html>
+  <body>
+    <button id="play-audio-btn">Play audio</button>
+    <script>
+      const audio_context = new AudioContext();
+      const oscillator = audio_context.createOscillator();
+      oscillator.connect(audio_context.destination);
+
+      const play_audio_btn = document.getElementById("play-audio-btn");
+
+      audio_context.addEventListener("statechange", () => {
+        console.log(audio_context.state);
+      });
+
+      play_audio_btn.addEventListener("click", () => {
+        oscillator.start();
+      });
+
+      window.addEventListener("load", () => {
+        navigator.audioSession.type = "auto";
+        console.log(audio_context.state);
+      });
+    </script> 
+  </body>
+</html>
+```
+
+## Web compatibility risks
+
+When `AudioContext.resume()` is called for an `AudioContext` in the `"closed"` state, the returned promise is rejected. With this proposal, the same behavior will happen when `AudioContext.resume()` is called while the `AudioContext` interrupted (could happen either during `"suspended"` or `"interrupted"`). In this case, a web page that is not aware of the existence of the "`interrupted`" state might think that the `AudioContext` has been closed.
+
+## Privacy considerations
+
+Moving the `AudioContexts` to the `"interrupted"` state might giveaway some information about the user's behavior - for example, when the user started a voice call or locked the screen. However, since "interrupting" an `AudioContext` could be associated with several types of interruptions, exposing this new state shouldn't be a major concern. Moreover, to mitigate privacy concerns, this proposal forbids automatically transitioning from `"suspended"` to `"interrupted"` to avoid unnecessarily exposing interruptions. This state transition will only happen if [`AudioContext.resume()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-resume) is called.
+
+## Considered alternatives
+
+This sections lists a number of alternatives taken into consideration prior to and during the writing of this document.
+
+### Re-use the `"suspended"` state
+
+We first considered in the [`"media-playback-while-not-visible"`](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/IframeMediaPause/iframe_media_pausing.md) permission policy proposal using the `"suspended"` state whenever media playback needed to be paused. However, this is not possible, because [only user code can suspend](https://webaudio.github.io/web-audio-api/#dom-audiocontext-suspended-by-user-slot) an `AudioContext`.
+
+## Stakeholder Feedback / Opposition
+
+- Chromium : No signals
+- WebKit : Positive

```

Comment:
Do we know if this proposal if interoperable with Safari's implementation?

Here's an interesting note from the WebAudio github issue:

"Christoph Guttandin mentioned that Safari sets the interrupted state if an audiocontext is silent and you switch tabs. When you switch back, it resumes. But if the context is playing, this doesn't happen. This all needs to be clarified."

https://github.com/WebAudio/web-audio-api/issues/2392#issuecomment-844465877